File: src/main/java/de/pflugradts/passbird/domain/service/eventhandling/EventRegistry.java
package de.pflugradts.passbird.domain.service.eventhandling;

import de.pflugradts.passbird.domain.model.ddd.AggregateRoot;
import de.pflugradts.passbird.domain.model.ddd.DomainEvent;

public interface EventRegistry {

    void register(AggregateRoot aggregateRoot);

    void register(DomainEvent domainEvent);

    void deregister(AggregateRoot aggregateRoot);

    void processEvents();

}

File: src/main/kotlin/de/pflugradts/passbird/domain/service/eventhandling/EventRegistry.kt
package de.pflugradts.passbird.domain.service.eventhandling

import de.pflugradts.passbird.domain.model.ddd.AggregateRoot
import de.pflugradts.passbird.domain.model.ddd.DomainEvent

interface EventRegistry {
    fun register(aggregateRoot: AggregateRoot)
    fun register(domainEvent: DomainEvent)
    fun deregister(aggregateRoot: AggregateRoot)
    fun processEvents()
}
